compiler/lib/subst.mli

```diff
@@ -39,3 +39,15 @@ val from_array : Var.t array -> Var.t -> Var.t
 val build_mapping : Var.t list -> Var.t list -> Var.t Var.Map.t
 
 val from_map : Var.t Var.Map.t -> Var.t -> Var.t
+
+(** The operations of this module also substitute the variables names that
+    appear on the left-hand-side of a {!constructor:Code.Let}, or as block
+    parameters, or as closure parameters, or are bound by an exception handler.
+    *)
+module Bound : sig
```

Comment:
I think we should move existing substitution function into another sub module to have a more regular API.  


```
val program : (Var.t -> Var.t) -> program -> program
val expr : (Var.t -> Var.t) -> expr -> expr
val instr : (Var.t -> Var.t) -> instr -> instr
val instrs : (Var.t -> Var.t) -> (instr * loc) list -> (instr * loc) list
val block : (Var.t -> Var.t) -> block -> block
val last : (Var.t -> Var.t) -> last * loc -> last * loc
val cont : (Var.t -> Var.t) -> int -> program -> program
```

Reply:
https://github.com/ocsigen/js_of_ocaml/pull/1461/commits/0a8183c79c7fb92afd20a0fdc836e5aeb4fb81c4